From: Marius Gripsgard <marius@ubports.com>
Date: Wed, 27 May 2020 20:39:52 +0200
Subject: [PATCH 128/194] usb: musb: sunxi: Handle MUSB_ULPI_BUSCONTROL

sunxi does not support setting ULPI buscontrol, it's used to set PHY to
use external vBus. This stubs out MUSB_ULPI_BUSCONTROL to be able to
handle save/restore events on suspend/resume events.
---
 drivers/usb/musb/sunxi.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/usb/musb/sunxi.c b/drivers/usb/musb/sunxi.c
index f3f76f2..9e47796 100644
--- a/drivers/usb/musb/sunxi.c
+++ b/drivers/usb/musb/sunxi.c
@@ -440,6 +440,10 @@ static u8 sunxi_musb_readb(void __iomem *addr, u32 offset)
 				return 0xde;
 
 			return readb(addr + SUNXI_MUSB_CONFIGDATA);
+
+		case MUSB_ULPI_BUSCONTROL:
+			return 0; /* ULPI buscontrol is not supported on sunix */
+
 		/* Offset for these is fixed by sunxi_musb_busctl_offset() */
 		case SUNXI_MUSB_TXFUNCADDR:
 		case SUNXI_MUSB_TXHUBADDR:
@@ -494,6 +498,12 @@ static void sunxi_musb_writeb(void __iomem *addr, unsigned offset, u8 data)
 			return writeb(data, addr + SUNXI_MUSB_TXFIFOSZ);
 		case MUSB_RXFIFOSZ:
 			return writeb(data, addr + SUNXI_MUSB_RXFIFOSZ);
+
+                case MUSB_ULPI_BUSCONTROL:
+                        if (data)
+                                dev_warn(sunxi_musb->controller->parent,
+                                        "sunxi-musb does not have ULPI buscontrol\n");
+                        return;
 		/* Offset for these is fixed by sunxi_musb_busctl_offset() */
 		case SUNXI_MUSB_TXFUNCADDR:
 		case SUNXI_MUSB_TXHUBADDR:
